### PR TITLE
[snowflake] quick start portion of the guide

### PR DESCRIPTION
**Pull request by jamiedemaria**
### Summary & Motivation
Splitting https://github.com/dagster-io/dagster/pull/10806 into a couple different PRs

This part of the guide is just the "quick start" section of the larger guide. it's been restructured a bit as well. let me know what you think. I did make many major changes to content yet, mostly playing with the structure

vercel - https://dagster-git-jamie-snowflake-quick-start-elementl.vercel.app/integrations/snowflake

### CODE REVIEW

**Comment by slopp** [comment]
I left some Vercel comments on the content while I was there, but this structure is awesome. I left a few structural comments that are small:

1. I think we can leave out the scary concepts section from pre-reqs and instead link to it at the end as a next step
2. The configure section should maybe incl the `build_snowflake_io_manager` part?
3. I'd suffix the two labels on the config tabs w/ Required and Optional 

The structure (and content) of defining tables vs assets and then using them is total goodness.

File: python_modules/dagster/dagster/_core/storage/db_io_manager.py [hunk 1]
```diff
@@ -87,9 +87,8 @@ def __init__(
         self._db_client = db_client
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster/_core/storage/db_io_manager.py:87]: revert. not sure how this got in from another pr

**Comment by jamiedemaria** [comment]
@slopp for 2:  there's now a `snowflake_pandas_io_manager` that you can directly import from `dagster_snowflake_pandas` I'm going to use that in the guide by default until more types are supported and then add a section describing the different IO managers to choose from

File: docs/content/integrations/snowflake.mdx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        resource_defs={
+            "io_manager": snowflake_pandas_io_manager.configured(
+                {
+                    "account": "abc1234.us-east-1", # required
+                    "user": {"env": "SNOWFLAKE_USER"}, # required
+                    "password": {"env": "SNOWFLAKE_PASSWORD"}, # required
+                    "role": "reader", # optional
+                    "warehouse": "PLANTS", # optional
+                    "database": "FLOWERS", # required
+                    "schema": "IRIS", # optional
```
Comment (rexledesma) [inline docs/content/integrations/snowflake.mdx]: This should be organized by required and optional.

File: docs/content/integrations/snowflake.mdx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+To store data in Snowflake using the Snowflake I/O manager, the definitions of your assets don't need to change. You can tell Dagster to use the Snowflake I/O Manager in your repository, and Dagster will handle storing and loading your assets in Snowflake.
+
+```python file=/integrations/snowflake/basic_example.py
+import pandas as pd
+from dagster_snowflake_pandas import snowflake_pandas_io_manager
+
+from dagster import asset, repository, with_resources
+
+
+@asset
```
Comment (rexledesma) [inline docs/content/integrations/snowflake.mdx:90]: explicitly tag the io_manager here, don't use the magic default `io_manager`

File: docs/content/integrations/snowflake.mdx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+### Make an existing table available in Dagster
+
+You may already have tables in Snowflake that you want to make available to other Dagster assets. You can create [source assets](/concepts/assets/software-defined-assets#defining-external-asset-dependencies) for these tables. By creating a source asset for the existing table, you tell Dagster how to find the table, so it can be fetched for downstream assets.
+
+```python file=/integrations/snowflake/source_asset.py
+from dagster_snowflake_pandas import snowflake_pandas_io_manager
+
+from dagster import SourceAsset, repository, with_resources
+
+daffodil_dataset = SourceAsset(key="daffodil_dataset")
```
Comment (rexledesma) [inline docs/content/integrations/snowflake.mdx]: we should only show this code in the snippet - the other stuff is extra.

**Review by rexledesma — CHANGES_REQUESTED** [review]

**Comment by yuhan** [comment]
i like this quickstart - it's short and neat. left a few comments in the preview but i think it's a good shape to push out!

im leaving this open for a bit to others a chance to comment : )

**Comment by jamiedemaria** [comment]
@erinkcochran87 you can do either or both, the main distinction is that one is for creating new tables that will be maintained/updated by dagster and one is for giving dagster access to existing tables that may be updated by external tools. I'll add some explanation before the tab group explaining this

**Review by yuhan — APPROVED** [review]
🚢

**Review by erinkcochran87 — APPROVED** [review]
:shipit:

**Pull request merged**